misc/sieve_of_eratosthenes.c

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @brief Get list of prime numbers using [Sieve of
+ * Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes)
+ * @details
+ * Sieve of Eratosthenes is an algorithm that finds all the primes
+ * between 2 and N.
+ *
+ * Time Complexity  : \f$O(N \cdot\log \log N)\f$
+ * <br/>Space Complexity : \f$O(N)\f$
+ *
+ */
+
+#include <assert.h> /// for 'assert' in 'test' function
+#include <stdbool.h> /// for 'bool' type in 'sieve' and 'test' function
+#include <stdlib.h> /// for 'calloc' in 'sieve' function
+#include <string.h> /// for 'memset' in 'sieve' function
+
+/**
+ * Return all primes between 2 and the given number
+ * @param N the largest number to be checked for primality
+ * @return is_prime a dynamically allocated array of `N + 1` booleans identifying if `i`^th number is a prime or not
+ */
+bool* sieve(int N)
+{
+    bool* primep = calloc(N+1, 1);
```

Comment:
dynamically allocated memory **must** be freeed.
In this case, a note should be added to the function that the function `free()` must be called on the output pointer after its use. 

For example, on line# 48, pointer to a new memory is returned. Hence, before the end of that function, there must be a `free(primep);`